pkg/argocd/update.go

```diff
@@ -551,13 +551,24 @@ func mergeHelmOverride(t *helmOverride, o *helmOverride) {
 }
 
 func mergeKustomizeOverride(t *kustomizeOverride, o *kustomizeOverride) {
-	for _, image := range *o.Kustomize.Images {
-		idx := t.Kustomize.Images.Find(image)
-		if idx != -1 {
-			(*t.Kustomize.Images)[idx] = image
-			continue
+	for _, newImage := range *o.Kustomize.Images {
+		found := false
+		newContainerImage := image.NewFromIdentifier(string(newImage))
+		for idx, existingImage := range *t.Kustomize.Images {
+			existingContainerImage := image.NewFromIdentifier(string(existingImage))
+			if newContainerImage.ImageName == existingContainerImage.ImageName &&
+				newContainerImage.RegistryURL == existingContainerImage.RegistryURL {
+				found = true
+				if existingContainerImage.ImageTag == nil ||
+					!(existingContainerImage.ImageTag).Equals(newContainerImage.ImageTag) {
```

Comment:
From what I see, at least from the tests that I'm trying to run, the new image tag can be nil and so the newImage gets added

Reply:
In real applications, new image tag is unlikely to be nil. If it is, something already went wrong earlier.

But if the new tag is nil, it will cause nil pointer dereferece and panic. We could add a nil check to prevent that from happening:

```go
if existingContainerImage.ImageTag == nil ||
(newContainerImage.ImageTag != nil && !(existingContainerImage.ImageTag).Equals(newContainerImage.ImageTag)) {
```